### ISSUE TITLE
I ran glooctl create upstream kube -i and it let me get two questions answered then errored

### ISSUE THREAD

**Issue by DaleBingham** [solo-io/gloo#206]
I ran
```
glooctl create upstream kube -i
```
and answered the namespace and name questions and then it errored with this -- 
```
Error: interactive mode not currently available for type kube
```

I would think it would give that right away?

**Comment by ilackarms** [comment]
to fix this, we can modify here:
https://github.com/solo-io/gloo/blob/master/projects/gloo/cli/pkg/cmd/create/upstream.go#L47

we want to inform the `AddUpstreamFlagsInteractive` of whether it's being called at the root `upstream` function or in a subcommand. it can then pick up the upstream type from a parameter

**Comment by rickducott** [comment]
Cheap fix here #363
